client/components/GenericTable/index.ts

```diff
@@ -0,0 +1,6 @@
+import GenericTable from './GenericTable';
+import HeaderCell from './HeaderCell';
+
+export default Object.assign(GenericTable, {
```

Comment:
Curious as to why you used `Object.assign` here 🤓 Is it just a shortcut to writing `GenericTable.HeaderCell = HeaderCell;` in the line above? Or is there another reason?

Reply:
Oh boy, you are the right guy to ask me this one. `GenericTable` has a function type (kind of `(props: any) => ReactElement`) defined at `'./GenericTable'`. It doesn't provide any mechanism to extend it while changing the concrete value: `GenericTable.HeaderCell = HeaderCell ;` would trigger an error for me because `GenericTable`'s type is not an extension of `{ HeaderCell: (props: any) => ReactElement; }`. I've noticed that `Object.assign()` intersects the types of its arguments, so we get `{ (props: any): ReactElement; HeaderCell: (props: any) => ReactElement; }` at the return type.

Reply:
Got it! 

Does this generate a side-effect? I mean, if someone imported from the `./GenericTable` file directly, the `GenericTable` exported there would also have the `HeaderCell` object due to `Object.assign`?

Of course, that wouldn't be a problem; TS wouldn't consider the original object to contain `HeaderCell` and I guess even the usual `console.log` would not print it there, it could only be confusing if you tried to see `GenericTable`'s keys with `Object.keys` or other similar functions. Just asking, again, out of curiosity 😛